http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandlerSpec.scala

```diff
@@ -375,19 +382,184 @@ class AzurePubsubHandlerSpec
+    when(vmReturn.powerState()).thenReturn(PowerState.STOPPED)
+
+    val passAzureVmService = new FakeAzureVmService {
+      override def startAzureVm(name: InstanceName, cloudContext: AzureCloudContext)(implicit
+        ev: Ask[IO, TraceId]
+      ): IO[Option[Mono[Void]]] = IO.some(Mono.empty[Void]())
+    }
+
+    val queue = QueueFactory.asyncTaskQueue()
+
+    val azureInterp =
+      makeAzureInterp(asyncTaskQueue = queue, azureVmService = passAzureVmService)
+
+    val res = for {
+      disk <- makePersistentDisk().copy(status = DiskStatus.Ready).save()
+      azureRuntimeConfig = RuntimeConfig.AzureConfig(MachineTypeName(VirtualMachineSizeTypes.STANDARD_A1.toString),
+                                                     disk.id,
+                                                     azureRegion
+      )
+      runtime = makeCluster(1)
+        .copy(
+          cloudContext = CloudContext.Azure(azureCloudContext)
+        )
+        .saveWithRuntimeConfig(azureRuntimeConfig)
+
+      // Assertions for runtime
+      assertions = for {
+        getRuntimeOpt <- clusterQuery.getClusterById(runtime.id).transaction
+        getRuntime = getRuntimeOpt.get
+      } yield getRuntime.status shouldBe RuntimeStatus.Running
+
+      // Mocked response, should succeed into the Some(mono) case
+      asyncTaskProcessor = AsyncTaskProcessor(AsyncTaskProcessor.Config(10, 10), queue)
+      _ <- azureInterp.startAndMonitorRuntime(runtime, azureCloudContext)
+
+      _ <- withInfiniteStream(asyncTaskProcessor.process, assertions)
+
+    } yield ()
+
+    res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+  }
+
+  it should "fail on startAzureVm - Azure runtime starting error" in isolatedDbTest {
+    val vmReturn = mock[VirtualMachine]
+    when(vmReturn.powerState()).thenReturn(PowerState.STOPPED)
+
+    val failAzureVmService = new FakeAzureVmService {
+      override def startAzureVm(name: InstanceName, cloudContext: AzureCloudContext)(implicit
+        ev: Ask[IO, TraceId]
+      ): IO[Option[Mono[Void]]] = IO.none
+    }
+
+    val queue = QueueFactory.asyncTaskQueue()
+
+    val azureInterp =
+      makeAzureInterp(asyncTaskQueue = queue, azureVmService = failAzureVmService)
+
+    val res = for {
+      disk <- makePersistentDisk().copy(status = DiskStatus.Ready).save()
+      azureRuntimeConfig = RuntimeConfig.AzureConfig(MachineTypeName(VirtualMachineSizeTypes.STANDARD_A1.toString),
+                                                     disk.id,
+                                                     azureRegion
+      )
+      runtime = makeCluster(1)
+        .copy(
+          cloudContext = CloudContext.Azure(azureCloudContext)
+        )
+        .saveWithRuntimeConfig(azureRuntimeConfig)
+
+      _ <- azureInterp.startAndMonitorRuntime(runtime, azureCloudContext)
+
+    } yield ()
+    a[AzureRuntimeStartingError] should be thrownBy {
+      res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
```

Comment:
this will never be thrown...

`res` is an `IO`, so you need to run it with `res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)`


And this is where I'd like the `handleError` function being tested instead of validating an exception is thrown

Reply:
When I run it without the `a[AzureRuntimeStartingError] should be thrownBy` wrapper I get a test failed.